src/Control/PathfinderRequestHandler.php

```diff
@@ -58,4 +116,451 @@ public function getController()
     {
         return $this->controller;
     }
+
+    /**
+     * @return ProgressStore
+     */
+    public function getStore()
+    {
+        if (!$this->store) {
+            $this->store = Injector::inst()->get(ProgressStore::class);
+
+            if (!$this->store) {
+                throw new \Exception('Unable to control Pathfinders without a configured store for progress');
+            }
+        }
+
+        return $this->store;
+    }
+
+    /**
+     * @return Question|null
+     */
+    public function getCurrentQuestion()
+    {
+        $questionId = (int) $this->getRequest()->getVar('id');
+
+        if (!$questionId) {
+            return null;
+        }
+
+        $stepNum = $this->getCurrentStepNumber();
+
+        if (!$stepNum) {
+            return null;
+        }
+
+        $question = $this->Questions()->byId($questionId);
+
+        if ($stepNum == 1) {
+            // Check we're on the first question
+            if ($question->ID !== $this->Questions()->first()->ID) {
+                return null;
+            }
+
+            // This is the first question
+            return $question;
+        }
+
+        $store = $this->getStore();
+
+        if ($store->canValidateSequence()) {
+            if (!$store->isInSequence($question, $stepNum)) {
+                // The requested question doesn't seem like the right one
+                return null;
+            }
+        }
+
+        return $question;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCurrentStepNumber()
+    {
+        return $this->getRequest()->getVar('step');
+    }
+
+    /**
+     * @return Form
+     */
+    public function Form()
+    {
+        $fields = FieldList::create(
+            HiddenField::create(
+                'CurrentQuestionID',
+                null,
+                $this->getCurrentQuestion() ? $this->getCurrentQuestion()->ID : null
+            ),
+            HiddenField::create('Step', null, $this->getCurrentStepNumber())
+        );
+
+        $actions = FieldList::create(
+            FormAction::create('goBack', 'Previous')
+                ->addExtraClass('action-next')
+                ->setUseButtonTag(true)
+                ->setDisabled(!$this->hasPreviousQuestion()),
+            FormAction::create('doSubmitQuestion', 'Next')
+                ->addExtraClass('action-prev')
+                ->setUseButtonTag(true)
+        );
+
+        $form = Form::create($this, 'Form', $fields, $actions);
```

Comment:
I think we will need to be able to set `disableSecurityToken()` on the form to prevent a session being created that will affect HTTP caching.

Maybe add it as a configurable option?

Reply:
Is the existing `updateForm` extension point sufficient for that? Or would you recommend making it configurable too?

Reply:
Yeah that will do actually mate. Nice one.